api/src/controllers/team.js

```diff
@@ -70,25 +74,47 @@ async function getTeam(req, res) {
 }
 
 /**
- * Gets the team document from the database on /team.
+ * Handle login request from /team/login
  * @param {*} req request object, containing team email and password in the body as JSON
- * @param {*} res response object, the found team document
+ * @param {*} res response object, the found teamId
  * @returns 200: the team was found
  * @returns 404: team is not found
  */
 async function loginTeam(req, res) {
-  Team.findOne({ email: req.body.email })
-    .then((team) => {
-      if (team == null) {
-        res.status(400).send('User not found');
-      } else if (team.password != req.body.password) {
-        res.status(403).send('Incorrect password');
-      } else {
-        console.log(team);
-        res.send(JSON.stringify({ team: team }));
-      }
-    })
-    .catch((err) => res.status(400).json('Error: ' + err));
+  try {
+    const foundTeam = await Team.findOne({ email: req.body.email })
+    if (!foundTeam) {
+      return res.status(400).send('Incorrect email/password'); // user not found 
+    } 
+    if (await bcrypt.compare(req.body.password, foundTeam.password)){
+      const teamObj = foundTeam.toObject(); // converts a mongoose object to a plain object 
+      // remove sensitive data 
+      delete teamObj.password 
+      const accessToken = jwt.sign(teamObj, process.env.JWT_SECRET_1 || "JWT_SECRET_1", {
+        expiresIn: '15m'
```

Comment:
Do you have a specific reason for the chosen expiry times for access and refresh tokens? I think the access token can probably be valid for longer like 1 hour cause what if they have a lot of publications to add manually? Also see the second link below.

Was reading out of curiosity and apparently [Google](https://cloud.google.com/apigee/docs/api-platform/antipatterns/oauth-long-expiration) suggests that refresh tokens should expire a bit after the access token, but then [this page](https://blog.cloud-elements.com/oauth-2-0-access-refresh-token-guide) shows otherwise.

Reply:
After the access token expired, a request along with the refresh token will be sent to the backend to get a new access token. 
So a user will be forced to log in again only if the refresh token expires (or cookies expires as we are using cookies as well). 
We can decide and change the expires times later stage cause I am thinking to make them as short as possible to test it in the next ticket.